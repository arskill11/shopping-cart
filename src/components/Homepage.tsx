import styled from 'styled-components';
import Goods from './HomepageGoods';
import Delivery from './HomepageDelivery';
import Info from './HomepageInfo';
import useAPIProducts from '../useAPIproducts';

const StyledHomepage = styled.div`
  display: flex;
  flex-direction: column;
`;

const Homepage = () => {
  // const goods: PropsAPI[] = useOutletContext();
  const goods = useAPIProducts('https://fakestoreapi.com/products?limit=4');
  console.log(goods);

  return (
    <StyledHomepage>
      <Info />
      <Goods goods={goods} />
      <Delivery />
    </StyledHomepage>
  );
};

export default Homepage;
