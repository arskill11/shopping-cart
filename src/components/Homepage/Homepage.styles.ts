import styled from 'styled-components';

const StyledHomepage = styled.div`
  display: flex;
  flex-direction: column;
`;

export default StyledHomepage;
