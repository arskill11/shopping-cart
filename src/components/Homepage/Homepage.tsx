import StyledHomepage from './Homepage.styles';
import { HomepageDelivery } from '../HomepageDelivery';
import { HomepageGoods } from '../HomepageGoods';
import { HomepageInfo } from '../HomepageInfo';
import useProductsQuery from '../../shared/hooks/useProductsQuery';

export const Homepage = () => {
  const goods = useProductsQuery('https://fakestoreapi.com/products?limit=4');

  return (
    <StyledHomepage>
      <HomepageInfo />
      <HomepageGoods goods={goods} />
      <HomepageDelivery />
    </StyledHomepage>
  );
};
