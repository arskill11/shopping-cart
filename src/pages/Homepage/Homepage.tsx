import { StyledHomepage } from './Homepage.styles';
import { HomepageDelivery } from '../../components/HomepageDelivery';
import { HomepageGoods } from '../../components/HomepageGoods';
import { HomepageInfo } from '../../components/HomepageInfo';
import useProductsQuery from '../../shared/hooks/useProductsQuery';
import { getLimitedProducts } from '../../api/products';

export const Homepage = () => {
  const goods = useProductsQuery(getLimitedProducts, '');

  return (
    <StyledHomepage>
      <HomepageInfo />
      <HomepageGoods goods={goods} />
      <HomepageDelivery />
    </StyledHomepage>
  );
};
