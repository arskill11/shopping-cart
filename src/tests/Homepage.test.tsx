import { render, screen } from '@testing-library/react';
import { describe, it, expect } from 'vitest';
import Homepage from '../components/Homepage';
import userEvent from '@testing-library/user-event';
