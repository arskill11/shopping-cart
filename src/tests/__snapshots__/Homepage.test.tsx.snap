// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Homepage > renders correctly 1`] = `
<div>
  <div
    class="sc-dcJtft dwqEEc"
  >
    <div
      class="sc-fqkwJk UZDdN"
    >
      <div
        class="blur-background"
      >
        <div
          class="shopInfo"
        >
          <h1>
            FakeShop
          </h1>
          <p>
            This is a fake shop to sell some goods not sindg
          </p>
        </div>
      </div>
    </div>
    <div
      class="sc-gEvDqW kvkVsY"
    >
      <h2>
        Our Featured Products!
      </h2>
      <div
        class="cards"
      />
    </div>
    <div
      class="sc-eqUzNf egLVeV"
    >
      <img
        alt=""
        src="/src/assets/deliveryTruck2.png"
      />
      <div>
        <h2>
          Free Delivery
        </h2>
        <p>
          We ship our goods for free regardlessly from your order sum
        </p>
      </div>
    </div>
  </div>
</div>
`;
